sqlmesh/core/engine_adapter/base.py

```diff
@@ -1567,6 +1739,35 @@ def _truncate_table(self, table_name: TableName) -> None:
         table = exp.to_table(table_name)
         self.execute(f"TRUNCATE TABLE {table.sql(dialect=self.dialect, identify=True)}")
 
+    def _create_comments(
+        self,
+        table_name: TableName,
+        table_comment: t.Optional[str],
+        column_comments: t.Optional[t.Dict[str, str]],
+        table_kind: str = "TABLE",
+    ) -> None:
+        """
+        Executes commands to create table and column comments.
+        """
+        if table_comment:
+            self.execute(
+                exp.Comment(
+                    this=exp.to_table(table_name),
+                    kind=table_kind,
+                    expression=exp.Literal.string(table_comment),
+                )
+            )
+
+        if column_comments:
+            for col, comment in column_comments.items():
+                self.execute(
+                    exp.Comment(
+                        this=exp.to_column(f"{table_name}.{col}"),
```

Comment:
This looks a bit iffy, if `table_name` is a `Table` expression then this silently invokes `.sql` on it which can produce unexpected SQL if there are quotes

Reply:
Yes - thanks for raising this. What is the correct/safe way to build a comment from table and column names in this context?

Reply:
You could create a table expression first like on L1755 and then construct the column expression by manually setting its `table`, `db` and `catalog` args to the corresponding args of the table.

@tobymao do we already have a helper for this in sqlmesh? I don't recall having something in sqlglot.

Reply:
what exactly are you trying to do?

Reply:
Build an expression that renders like this with the right identifier quoting: `COMMENT ON COLUMN my_table.my_column IS 'column comment'`

Reply:
do this at the top

table = exp.to_table(...)

then you can do

exp.column(col, *reverse(table.parts))